src/pods/test-multiple-choice/test-multiple-choice.container.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import { TestMultipleChoiceComponent } from './test-multiple-choice.component';
+import { Verb, createDefaultVerb } from 'common/model';
+import { globalVerbsContext } from 'core/verbs';
+import { pickRandomVerb } from './test-multiple-choice.business';
+import { useHistory } from 'react-router-dom';
+import { routes } from 'core/router';
+import { scoreContext } from 'core/score';
+import { settingsContext } from 'core/settings';
+
+const INITIAL_ANSWERED_CORRECTLY = 0;
+const INITIAL_CURRENT_QUESTION = 1;
+
+export const TestMultipleChoiceContainer = () => {
+  const history = useHistory();
+  const { selectedVerbs, verbCollection } = React.useContext(globalVerbsContext);
+  const { setScore } = React.useContext(scoreContext);
+  const { userSettings } = React.useContext(settingsContext);
+  const [totalQuestions] = React.useState(userSettings.numberQuestions);
+
+  const [currentQuestion, setCurrentQuestion] = React.useState(
+    INITIAL_CURRENT_QUESTION
+  );
+  const [currentVerb, setCurrentVerb] = React.useState<Verb>(
+    createDefaultVerb()
+  );
+  const [otherOption1, setOtherOption1] = React.useState<Verb>(
+    createDefaultVerb()
+  );
+  const [otherOption2, setOtherOption2] = React.useState<Verb>(
+    createDefaultVerb()
+  );
+  const [currentScore, setCurrentScore] = React.useState(
+    INITIAL_ANSWERED_CORRECTLY
+  );
+
+  React.useEffect(() => {
+    setScore({ totalQuestions, answeredCorrectly: INITIAL_ANSWERED_CORRECTLY });
+  }, []);
+
+  React.useEffect(() => {
+    if (currentQuestion <= totalQuestions) {
+      const randomVerb = pickRandomVerb(selectedVerbs, verbCollection);
+      setCurrentVerb(randomVerb);
+      let otherRandom:Verb, otherRandom2:Verb;
+      // Check the first other option is not repeated
+      do {
+        otherRandom = pickRandomVerb(selectedVerbs, verbCollection);
+      } while(randomVerb.infinitive === otherRandom.infinitive);
+      setOtherOption1(otherRandom);
+      // Check the second other option is not repeated
+      do {
+        otherRandom2 = pickRandomVerb(selectedVerbs, verbCollection);
+      } while(randomVerb.infinitive === otherRandom2.infinitive ||
+              otherRandom.infinitive === otherRandom2.infinitive);
+      setOtherOption2(otherRandom2);
+    }
```

Comment:
This code can be extracted to a _test-multiple-choice.business.tsx_ it could be something like:

_test-multiple-choice.container.tsx_
```tsx
React.useEffect(() => {
  const randomVerb = pickRandomVerb(selectedVerbs, verbCollection);
  setCurrentVerb(randomVerb);
  const otherVerbOptions = pickFalseVerbOptions(rightVerbOption);
  setOtherOption1(otherVerbOptions[1]);
  setOtherOption2(otherVerbOptions[2]);
},[currentQuestion]
```

Pseudocode, add unit tests to this

_test-multiple-choice.business.ts_
```ts
const pickFalseVerbOptions = (rightVerbOption : string) => {
   const optionA = [pickOption(rightVerbOption)];
   const optionB = [optionA, pickOption(rigthVerbOption)];

  return [optionA, optionB];
}

const pickOption = (optionsExcluded : string[]) : string => {
      do {
        selectedOption = pickRandomVerb(selectedVerbs, verbCollection);
      } while(optionsExcluded.findIndex(verb => verb === selectedOption.infinitive) != -1);

    return selectedOption;
}
```

Then we can add unit tests to the business part

Reply:
I've moved the function to business file. I've also created a spec file on each pod in order to test business files